server/src/main/java/org/candlepin/controller/ProductManager.java

```diff
@@ -511,6 +511,8 @@ private Product applyProductChanges(Product entity, ProductData update, Owner ow
             entity.setDependentProductIds(update.getDependentProductIds());
         }
 
+        entity.setLocked(update.isLocked());
```

Comment:
Just remembered, we need a null check on update.isLocked, in the event there's no change to the value.

```
if (update.isLocked() != null) {
    entity.setLocked(update.isLocked());
}
```